### DISCUSSION TITLE
SECRET MANAGEMENT

### DISCUSSION THREAD

**Discussion by compgeniuses** [coollabsio/coolify#1692] [Feature Requests]
IN RELATION TO THIS: https://github.com/Infisical/infisical/issues/1350

I was wodering if coolify would provide secrets manageemnt functions with possibility of integration to infisical for the secrets management.

https://infisical.com/

this would provide a more coherent secrets management

infisical is also opensource and self hostable, and could be a service within coolify

**Comment by rennokki** [comment]
This was added some time ago.

**Reply by Mpic** [reply]
Where is it ?

**Reply by janbiasi** [reply]
> This was added some time ago.

I think the question isn't related to the deployment of infisical itself, but using infisical to manage secrets for coolify within infisical instead of coolify.

**Comment by Adee1499** [comment]
Just wondering if secret management via Infisical (or similar providers), where Coolify applications can pull secrets directly into environment variables at runtime, is on the roadmap?

The goal would be to manage secrets entirely in Infisical and let Coolify consume them, rather than having to store them separately inside Coolify. This would improve both security and ease of management.

Is this kind of integration planned or would it make sense as a feature request?

**Reply by compgeniuses** [reply]
i actually have a change of heart and would prefer to use OpenBao instead

**Comment by goat255** [comment]
This feature is much needed.

**Comment by drpark** [comment]
Really really needed

**Comment by julianbrennum** [comment]
Do we know whether this is being considered? It would be really helpful if we could use Infisical for secrets in Coolify.

**Comment by morandalex** [comment]
this is the only feature missing that decides if I start using coolify or not. now I am handling docker compose manually because of this.. @andrasbacsai can you check this issue ?

**Comment by andrasbacsai** [comment]
Yes i would like to have something like this, but not sure about which providers should be supported in the first place.

**Comment by morandalex** [comment]
@andrasbacsai check what are doing in mise with fnox 
https://github.com/jdx/fnox

they have a really interesting flexible approach on how to use secrets

**Reply by morandalex** [reply]
hi @andrasbacsai did you check this approach ?

**Reply by janbiasi** [reply]
there's also https://secretspec.dev which provides an SDK for PHP as well

**Comment by djsisson** [comment]
Docker are introducing a secrets engine where it will pass any env starting with se:// over to a secrets plugin for whichever provider you want to use
https://github.com/docker/secrets-engine
